File: backend/TravelApp.Api/Models/Destination.cs
namespace TravelApp.Api.Models;

public class Destination
{
    public int Id { get; set; }
    public int TravelPlanId { get; set; }
    public TravelPlan TravelPlan { get; set; } = null!;

    public string Name { get; set; } = string.Empty;
    public string Location { get; set; } = string.Empty;
    public DateTime StartDate { get; set; }
    public DateTime EndDate { get; set; }
    public string? Description { get; set; }
    public string? Notes { get; set; }
    public int SortOrder { get; set; }
}

File: frontend/src/models/Destination.js
export function toDestination(data) {
  return {
    id: Number(data.id),
    name: data.name || '',
    location: data.location || '',
    startDate: data.startDate || '',
    endDate: data.endDate || '',
    description: data.description || null,
    notes: data.notes || null,
    sortOrder: Number(data.sortOrder || 0),
  };
}
